runtime/src/iree/hal/drivers/cuda/nccl_channel.c

```diff
@@ -294,6 +294,36 @@ static iree_status_t iree_hal_cuda_nccl_submit_batch_entry(
           "ncclAllReduce");
       break;
     }
+    case IREE_HAL_COLLECTIVE_KIND_ALL_TO_ALL: {
+      CUdeviceptr sendbuff =
+          iree_hal_cuda_buffer_device_pointer(
+              iree_hal_buffer_allocated_buffer(entry->send_binding.buffer)) +
+          iree_hal_buffer_byte_offset(entry->send_binding.buffer) +
+          entry->send_binding.offset;
+      CUdeviceptr recvbuff =
+          iree_hal_cuda_buffer_device_pointer(
+              iree_hal_buffer_allocated_buffer(entry->recv_binding.buffer)) +
+          iree_hal_buffer_byte_offset(entry->recv_binding.buffer) +
+          entry->recv_binding.offset;
+      iree_device_size_t send_count = entry->element_count / channel->count;
+      iree_device_size_t element_size_bytes;
+      IREE_RETURN_IF_ERROR(iree_hal_collective_element_byte_count(
+          entry->op.element_type, &element_size_bytes));
+      iree_device_size_t rank_offset = send_count * element_size_bytes;
+      NCCL_RETURN_IF_ERROR(syms, ncclGroupStart(), "ncclGroupStart");
```

Comment:
the group start and end aren't required because we always put them around the collective batches we submit:
https://github.com/openxla/iree/blob/c7925912b2f76b34335ab3d6949cd87a0c4f6071/runtime/src/iree/hal/drivers/cuda/nccl_channel.c#L411-L418

(nested group start/end are no-ops - if they're useful for some logging or something we could keep them but otherwise find to remove)